src/util.c

```diff
@@ -83,3 +83,43 @@ void streamStr(Stream *buf, const char *str)
 {
     streamMem(buf, str, strlen(str));
 }
+
+#if !defined(HAVE_CLOCK_NANOSLEEP)
+#include <sys/time.h>
+
+/* clock_nanosleep: clock_nanosleep for backwards systems lacking this function
+ *
+ * POSIX says that if clock is CLOCK_REALTIME and CLOCK_REALTIME is set while
+ * clock_nanosleep() waits on an absolute time, the new value should determine
+ * the moment clock_nanosleep returns. This implementation does not comply with
+ * that.
+ */
+int
+clock_nanosleep(clockid_t clock, int flags, const struct timespec *timeout,
```

Comment:
Shadowing a standard function with an incomplete implementation seems like a bad idea. Instead I'd just implement it minimally inside a `#ifdef` in `scrotWaitUntil()`. (I'm not a fan of `#ifdef`s inside of functions but it's the lesser of two evil here).

Reply:
P.S: Another alternative:

```c
#if defined(HAVE_CLOCK_NANOSLEEP)
scrotWaitUntil() { ... }
#else
scrotWaitUntil() { ... } // fallback implementation
#endif
```

Reply:
That's a deliberate design choice, I think it makes it easier to understand the code. 

Anyone who knows the standard `clock_nanosleep()` can understand what `scrotWaitUntil()` is doing without having to reason about `#ifdef`. Someone who wants to go deeper into the code can read my fallback `clock_nanosleep()` implementation which is compliant save for the case where the call is `clock_nanosleep(CLOCK_REALTIME, TIMER_ABSTIME, ...)`, but if the platforms doesn't have `clock_nanosleep()`, it most likely also doesn't have an equivalent function to use (OpenBSD does have  `__thrsleep()` but it's a private API), so that's a bug we can't avoid on those platforms if we were to add a sleep on CLOCK_REALTIME to scrot anyway.

Can't it stay as is?

Reply:
> I think it makes it easier to understand the code

It also adds unnecessary footgun into the codebase because people will need to be aware that the fallback path isn't fully compliant. No one's going to intuitively realize that we have a incomplete fallback and might end up using unimplemented feature.

Reply:
P.S: what makes standard function "easy to understand" is that they are standard, and thus should behave predictably across systems. But the moment someone adds an incomplete fallback, that predictability is lost.

Reply:
What if I make the fallback implementation call `err(1, "%s line %d: sleeping on an absolute CLOCK_REALTIME is unimplemented", __FILE__, __LINE__);` in case it's called with `clock_nanosleep(CLOCK_REALTIME, TIMER_ABSTIME, ...)`?